replica/table.cc

```diff
@@ -2523,6 +2523,25 @@ table::make_partition_presence_checker(lw_shared_ptr<const sstables::sstable_set
     };
 }
 
+max_purgeable_fn table::get_max_purgeable_fn_for_compacting_reader() const {
```

Comment:
This is too generic, a better name would be `get_max_purgeable_fn_for_cache_underlying_reader()`.
This name specifies that this is specific to the cache's underlying reader, which already tells you why sstables need not be considered (because all sstables that contain the read range are already included in the read).

Reply:
Done.